Update Annotation Collection Vertex Schema to Include "Tag" and "Layer" Fields and Remove provider-logging

michaelehab opened the pull request:
Fix #11 
# Changes
- Remove provider-logging
- Update Annotation Vertex Schema to include "Tag" and "Layer" fields
- Test these changes with Jenkins for both "app" and "cicd" layers
- Solve "gofmt" warnings emitted during "make test"
# Screenshots
## CICD Layer Annotation (Tested with [alvarium-pipelines example pipeline](https://github.com/project-alvarium/alvarium-pipelines/blob/main/README.md))
![cicd annotation](https://github.com/project-alvarium/scoring-apps-go/assets/29122581/63c51936-ad13-4525-9097-14867b53d64d)

## Application Layer Annotation (Tested with [ones-demo-2021](https://github.com/michaelehab/ones-demo-2021))
![application annotation](https://github.com/project-alvarium/scoring-apps-go/assets/29122581/fb8b13c4-abdd-4d91-b31f-2e9d508ab160)

tsconn23 reviewed (approved):
LGTM

tsconn23 reviewed (changes requested):
@michaelehab This PR looks good and I initially approved it. However I noticed you have an unverified commit in the history. That needs to be corrected b/c when I squash the history via the Github UI, it says this PR was "co-authored" by you. Try to squash your commit history locally and then update the PR. That should eliminate the unsigned commit.

michaelehab commented:
I've squashed the commits into a single verified commit and updated the PR. I apologize for any inconvenience caused by the earlier unverified commit.
